pennylane/interfaces/execution.py

```diff
@@ -76,8 +76,7 @@ def _adjoint_jacobian_expansion(
         non_trainable = ~qml.operation.is_trainable
 
     stop_at = ~qml.operation.is_measurement & (
-        non_trainable  # pylint: disable=unsupported-binary-operation
-        | qml.operation.has_unitary_gen
+        non_trainable | qml.operation.has_gen  # pylint: disable=unsupported-binary-operation

```

Comment:
Actually, I remeber why I used `qml.operation.has_unitary_gen` here instead of `qml.operation.has_gen`. It was a massive performance difference. `qml.operation.has_gen` relies on a try-except logic with exceptions that are not exceptional, so it gets *extremely* slow.

Reply:
With #3875 merged in, this should be okay now :)